backend/instancemgmt/instance_manager.go

```diff
@@ -96,9 +114,10 @@ func (im *instanceManager) Get(ctx context.Context, pluginContext backend.Plugin
 		return nil, err
 	}
 	// Double-checked locking for update/create criteria
-	im.locker.RLock(cacheKey)
-	item, ok := im.cache.Load(cacheKey)
-	im.locker.RUnlock(cacheKey)
+	strKey := fmt.Sprintf("%v", cacheKey)
+	im.locker.RLock(strKey)
+	item, ok := im.cache.Get(strKey)
+	im.locker.RUnlock(strKey)
```

Comment:
the shared read locking here looks unnecessary. even though your PR didn't introduce it, I'd be in favour of removing it.

Reply:
removed!